### PR TITLE
add doc describing how to create a release process

### PR DESCRIPTION
**Pull request by tobsan**
This article should be usable for anyone who wants to create their own release process.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: docs/articles/workflow/creating-release-process.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+process.
+
+It is important to document the process and the various steps in it. In order to
+be able to improve the actual process, and to keep it transparent we suggest
+that the release process is kept together with the rest of the documentation for
+the project.
+
+Create a list of artifacts
+--------------------------
+To be able to maintain a release, a list of artifacts is required to have. The
```
Comment (thenor) [inline docs/articles/workflow/creating-release-process.rst]: I don't understand this first sentence. What do you wish to communicate?

File: docs/articles/workflow/creating-release-process.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+We recommend having a file with release notes as one of the artifacts, so a user
+only needs one file to read up on the release contents.
+
+Create a list of component attributes for the release
+-----------------------------------------------------
+These attributes are mostly useful internally for being able to tie the various
+components of the release to the actual release. If components are stored in
+source code repositories, it makes sense to point out a specific revision of the
+source code, for example. Other types of attributes could be naming schemes or
+file name patterns, for example.
```
Comment (thenor) [inline docs/articles/workflow/creating-release-process.rst]: "If components are stored in source code repositories, it makes sense to point out a specific revision of the source code, for example." This sentence is well formed in speech, but not in text. I would rephrase it. Perhaps: "E.g. if components are stored in source code repositories, it makes sense to point out a specific revision of the source code."
Also the same for the last sentence. My suggestion is just to remove "for example".

**Review by thenor — CHANGES_REQUESTED** [review]
There are a few sentences I would like to rephrased for clarity.

File: docs/articles/workflow/creating-release-process.rst (hunk 2 above)
Comment (tobsan) [reply docs/articles/workflow/creating-release-process.rst]: Will do!

File: docs/articles/workflow/creating-release-process.rst (hunk 1 above)
Comment (tobsan) [reply docs/articles/workflow/creating-release-process.rst]: That it is important to know what artifacts are in a release. But I agree that it can be written in a nicer way.

**Review by thenor — APPROVED** [review]

File: docs/articles/workflow/creating-release-process.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
+:orphan:
+
+Creating a release process
+==========================
+
+In order to make sure the software that one releases is of high quality,
```
Comment (GordanM) [inline docs/articles/workflow/creating-release-process.rst]: "In order to ensure a high quality software release..."

File: docs/articles/workflow/creating-release-process.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+that the release process is kept together with the rest of the documentation for
+the project.
+
+Create a list of artifacts
+--------------------------
+Once released, it is important to know what was part of that specific release.
+For that, a list of artifacts is required. These artifacts constitute the
+product of the release, i.e. the matter that is released. This means that the
+recipient of the release should need nothing more than the artifacts (except
+readily available tooling etc) in order to be able to use the release.
```
Comment (GordanM) [inline docs/articles/workflow/creating-release-process.rst]: "... etc.)"

**Review by GordanM — APPROVED** [review]
Minor wording but overall LGTM.

**Comment by tobsan** [comment]
Thanks @GordanM, I've updated according to your comments.
